packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1344,11 +1340,22 @@ export class GarbageCollector implements IGarbageCollector {
         if (this.isSummarizerClient) {
             this.pendingEventsQueue.push({ ...propsToLog, usageType, state });
         } else {
-            this.mc.logger.sendErrorEvent({
-                ...propsToLog,
-                eventName: `${state}Object_${usageType}`,
-                pkg: packagePath ? { value: packagePath.join("/"), tag: TelemetryDataTag.CodeArtifact } : undefined,
-            });
+            // For non-summarizer clients, only log "Loaded" type events since these objects may not be loaded in the
+            // summarizer clients if they are based off of user actions (such as scrolling to content for these objects)
+            if (usageType === "Loaded") {
+                this.mc.logger.sendErrorEvent({
+                    ...propsToLog,
+                    eventName: `${state}Object_${usageType}`,
+                    pkg: packagePath ? { value: packagePath.join("/"), tag: TelemetryDataTag.CodeArtifact } : undefined,
+                });
+            }
+
+            // If AccessViolation Detection is enabed, close the main container
+            if (this.mc.config.getBoolean(accessViolationOnSweepReadyUsageKey) === true && state === "SweepReady") {
+                //* Create a proper error class
+                //* Put a prop on here indicating it's "best guess"
+                this.runtime.closeFn({ errorType: "accessViolationError", message: "SweepReady object was used! :(" });
```

Comment:
Are we not using something like ObjectDeletedByGCUsed? for error type? Or what Tony wrote.

Reply:
Still a draft - Just pushed updates here.